sys/include/net/dns/msg.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2021 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_dns_msg DNS message parser and composer
+ * @ingroup     net_dns
+ * @brief       Parsing and composition of DNS messages
+ * @{
+ *
+ * @file
+ * @brief       Definitions for parsing and composition of DNS messages
+ *
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+#ifndef NET_DNS_MSG_H
+#define NET_DNS_MSG_H
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup net_dns_msg_conf DNS message parsing and composition configuration
+ * @ingroup  config
+ * @{
+ */
+/**
+ * @brief   maximum DNS message length
+ */
+#ifndef CONFIG_DNS_MSG_LEN
+#define CONFIG_DNS_MSG_LEN         (128U)
+#endif  /* CONFIG_DNS_MSG_LEN */
+/** @} */
+
+/**
+ * @brief DNS internal structure
+ */
+typedef struct {
+    uint16_t id;        /**< read           */
+    uint16_t flags;     /**< DNS            */
+    uint16_t qdcount;   /**< RFC            */
+    uint16_t ancount;   /**< for            */
+    uint16_t nscount;   /**< detailed       */
+    uint16_t arcount;   /**< explanations   */
+    uint8_t payload[];  /**< !!             */
+} dns_hdr_t;
```

Comment:
eh, could at least have linked the relevant RFC :wink:

Reply:
This is just copied from @kaspar030's original source, but will make the doc a bit friendlier :-).